apps/rush/src/utilities/ShrinkwrapFile.ts

```diff
@@ -107,8 +111,20 @@ export default class ShrinkwrapFile {
       return false;
     }
 
-    // If we found it, the version must be compatible
-    return semver.satisfies(dependencyJson.version, versionRange);
+    const result: npmPackageArg.IResult = npmPackageArg.resolve(dependencyName, versionRange);
+    switch (result.type) {
+      case 'version':
+      case 'range':
+        // If it's a SemVer pattern, then require that the shrinkwrapped version must be compatible
+        return semver.satisfies(dependencyJson.version, versionRange);
+      default:
+        // Only warn once for each spec
+        if (!this._alreadyWarnedSpecs.has(result.rawSpec)) {
+          this._alreadyWarnedSpecs.add(result.rawSpec);
+          console.log(colors.yellow(`WARNING: Not validating ${result.type}-based spec: "${result.rawSpec}"`));
```

Comment:
I disagree.  Since the warning is about the specifier, I don't think we care if multiple packages use the same specifier.

---
In reply to: [125353176](https://github.com/Microsoft/web-build-tools/pull/260#discussion_r125353176) [](ancestors = 125353176)